src/EFCore.PG/Scaffolding/Internal/NpgsqlDatabaseModelFactory.cs

```diff
@@ -829,21 +831,46 @@ ELSE pg_get_expr(indpred, cls.oid)
             [CanBeNull] Func<string, string> schemaFilter,
             [NotNull] IDiagnosticsLogger<DbLoggerCategory.Scaffolding> logger)
         {
+            // Note: we consult information_schema.sequences instead of pg_sequence but the latter was only introduced
+            // in PG 10
             var commandText = $@"
 SELECT
-    nspname, relname,
-    format_type(typ.oid, 0) AS seqtype,
-    seqstart, seqmin, seqmax, seqincrement, seqcycle, seqcache
-FROM pg_class AS cls
-JOIN pg_namespace AS ns ON ns.oid = cls.relnamespace
-JOIN pg_sequence AS seq ON seq.seqrelid = cls.oid
-JOIN pg_type AS typ ON typ.oid = seq.seqtypid
+  sequence_schema, sequence_name,
+  data_type AS seqtype,
+  start_value::bigint AS seqstart,
+  minimum_value::bigint AS seqmin,
+  maximum_value::bigint AS seqmax,
+  increment::bigint AS seqincrement,
+  1::bigint AS seqcache,
+  CASE
+    WHEN cycle_option = 'YES' THEN TRUE
+    ELSE FALSE
+  END AS seqcycle
+FROM information_schema.sequences
+JOIN pg_namespace AS ns ON ns.nspname = sequence_schema
+JOIN pg_class AS cls ON cls.relnamespace = ns.oid AND cls.relname = sequence_name
 WHERE
   cls.relkind = 'S'
   /* AND seqtype IN ('integer', 'bigint', 'smallint') */
   /* Filter out owned serial and identity sequences */
   AND NOT EXISTS (SELECT * FROM pg_depend AS dep WHERE dep.objid = cls.oid AND dep.deptype IN ('i', 'I', 'a'))
   {(schemaFilter != null ? $"AND {schemaFilter("nspname")}" : null)}";
+//
+//            var commandText = $@"
+//SELECT
+//    nspname, relname,
+//    format_type(typ.oid, 0) AS seqtype,
+//    seqstart, seqmin, seqmax, seqincrement, seqcycle, seqcache
+//FROM pg_class AS cls
+//JOIN pg_namespace AS ns ON ns.oid = cls.relnamespace
+//JOIN pg_sequence AS seq ON seq.seqrelid = cls.oid
+//JOIN pg_type AS typ ON typ.oid = seq.seqtypid
+//WHERE
+//  cls.relkind = 'S'
+//  /* AND seqtype IN ('integer', 'bigint', 'smallint') */
+//  /* Filter out owned serial and identity sequences */
+//  AND NOT EXISTS (SELECT * FROM pg_depend AS dep WHERE dep.objid = cls.oid AND dep.deptype IN ('i', 'I', 'a'))
+//  {(schemaFilter != null ? $"AND {schemaFilter("nspname")}" : null)}";
```

Comment:
Do you need this comment? There is a SCM which tracks changes and you can find the original query in it.

Reply:
Nope, left it by mistake, thanks!